internal/service/encryptionatrest/model_encryption_at_rest.go

```diff
@@ -0,0 +1,120 @@
+			CustomerMasterKeyID: types.StringValue(awsKms.GetCustomerMasterKeyID()),
+			Region:              types.StringValue(awsKms.GetRegion()),
+			AccessKeyID:         conversion.StringNullIfEmpty(awsKms.GetAccessKeyID()),
+			SecretAccessKey:     conversion.StringNullIfEmpty(awsKms.GetSecretAccessKey()),
+			RoleID:              conversion.StringNullIfEmpty(awsKms.GetRoleId()),
+		},
+	}
+}
+
+func NewTFAzureKeyVaultConfig(ctx context.Context, az *admin.AzureKeyVault) []TfAzureKeyVaultConfigModel {
+	if az == nil {
+		return []TfAzureKeyVaultConfigModel{}
+	}
+
+	return []TfAzureKeyVaultConfigModel{
+		{
+			Enabled:           types.BoolPointerValue(az.Enabled),
+			ClientID:          types.StringValue(az.GetClientID()),
+			AzureEnvironment:  types.StringValue(az.GetAzureEnvironment()),
+			SubscriptionID:    types.StringValue(az.GetSubscriptionID()),
+			ResourceGroupName: types.StringValue(az.GetResourceGroupName()),
+			KeyVaultName:      types.StringValue(az.GetKeyVaultName()),
+			KeyIdentifier:     types.StringValue(az.GetKeyIdentifier()),
+			TenantID:          types.StringValue(az.GetTenantID()),
+			Secret:            conversion.StringNullIfEmpty(az.GetSecret()),
+		},
+	}
+}
+
+func NewTFGcpKmsConfig(ctx context.Context, gcpKms *admin.GoogleCloudKMS) []TfGcpKmsConfigModel {
+	if gcpKms == nil {
+		return []TfGcpKmsConfigModel{}
+	}
+
+	return []TfGcpKmsConfigModel{
+		{
+			Enabled:              types.BoolPointerValue(gcpKms.Enabled),
+			KeyVersionResourceID: types.StringValue(gcpKms.GetKeyVersionResourceID()),
+			ServiceAccountKey:    conversion.StringNullIfEmpty(gcpKms.GetServiceAccountKey()),
+		},
+	}
+}
+
+func NewAtlasAwsKms(tfAwsKmsConfigSlice []TfAwsKmsConfigModel) *admin.AWSKMSConfiguration {
+	if tfAwsKmsConfigSlice == nil || len(tfAwsKmsConfigSlice) < 1 {
+		return &admin.AWSKMSConfiguration{}
+	}
+	v := tfAwsKmsConfigSlice[0]
+
+	awsRegion, _ := conversion.ValRegion(v.Region.ValueString())
+
+	return &admin.AWSKMSConfiguration{
+		Enabled:             v.Enabled.ValueBoolPointer(),
+		AccessKeyID:         v.AccessKeyID.ValueStringPointer(),
+		SecretAccessKey:     v.SecretAccessKey.ValueStringPointer(),
+		CustomerMasterKeyID: v.CustomerMasterKeyID.ValueStringPointer(),
+		Region:              conversion.StringPtr(awsRegion),
+		RoleId:              v.RoleID.ValueStringPointer(),
+	}
+}
+
+func NewAtlasGcpKms(tfGcpKmsConfigSlice []TfGcpKmsConfigModel) *admin.GoogleCloudKMS {
+	if tfGcpKmsConfigSlice == nil || len(tfGcpKmsConfigSlice) < 1 {
+		return &admin.GoogleCloudKMS{}
+	}
+	v := tfGcpKmsConfigSlice[0]
+
+	return &admin.GoogleCloudKMS{
+		Enabled:              v.Enabled.ValueBoolPointer(),
+		ServiceAccountKey:    v.ServiceAccountKey.ValueStringPointer(),
+		KeyVersionResourceID: v.KeyVersionResourceID.ValueStringPointer(),
+	}
+}
+
+func NewAtlasAzureKeyVault(tfAzKeyVaultConfigSlice []TfAzureKeyVaultConfigModel) *admin.AzureKeyVault {
+	if tfAzKeyVaultConfigSlice == nil || len(tfAzKeyVaultConfigSlice) < 1 {
```

Comment:
```suggestion
	if len(tfAzKeyVaultConfigSlice) == 0 {
```

if it's nil, the len is 0, no need to check it apart,  also it think it's a bit more clear == 0 than < 1 although it's the same here